ASAA-32-Projects Screen

LuisRocketRangel opened the pull request:
Requirements:
-Add a call to get the Projects from Bitbucket
-Add a Card To Display a Project Item
-Add a screen to display the project
-Add a new route to navigate to the Project screen

Solution:
-Did above

LuisRocketRangel committed:
ASAA-32-Projects Screen

Requirements:
-Add a call to get the Projects from Bitbucket
-Add a Card To Display a Project Item
-Add a screen to display the project
-Add a new route to navigate to the Project screen

Solution:
-Did above

IvanTelisman commented on domain/src/main/java/com/bottlerocketstudios/brarchitecture/domain/models/Project.kt:
```diff
@@ -0,0 +1,9 @@
+    val name: String,
+    val key: String,
+    val updatedOn: ZonedDateTime?,
+)
```
Hey @LuisRocketRangel just in case for this domain model object to help differentiate between other model objects let's extend that DomainModel interface. Would look something like this: 

```
data class Project(
    val name: String,
    val key: String,
    val updatedOn: ZonedDateTime?,
) : DomainModel
```

IvanTelisman commented:
Looks good @LuisRocketRangel nice work on this one!

IvanTelisman commented on app/src/main/java/com/bottlerocketstudios/brarchitecture/ui/projects/ProjectsViewModel.kt:
```diff
@@ -0,0 +1,41 @@
+    // Init logic
+    init {
+        launchIO {
+            getProjects()
```
Hey @LuisRocketRangel , could probably just inline this function call in the init block here where you could get rid of that getProjects function below. Would just look something like:

```
init {
        launchIO {
            repo.getProjects()
        }
    }
```

LuisRocketRangel committed:
ASAA-32-PR Feedback(Ivan)

Requirements:
-DomainModel object change
-Inline repo.getProjects

Solution:
-Did above

LuisRocketRangel commented:
Fixed it!
